### PR TITLE
Prompt v3

### PR DESCRIPTION
**Pull request by odp**
Includes
* prompt v3
* search agent v3
* CustomLLMV3
* test v3

### CODE REVIEW

File: autoagents/agents/agents/search_v3.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        for i, (action, observation) in enumerate(intermediate_steps):
+            if action.tool not in [tool.name for tool in self.tools]:
+                raise Exception("Invalid tool requested by the model.")
+            parsed = json.loads(action.log)
+            if i == len(intermediate_steps) - 1:
+                # Add observation only for the last action
+                parsed["observation"] = observation
+            history.append(parsed)
+        self.ialogger.add_history(history)
+        kwargs["agent_scratchpad"] = json.dumps(history)
```
Comment (yukiontheiceberg) [inline autoagents/agents/agents/search_v3.py:75]: Do we still need to add keys into kwargs?

File: autoagents/agents/agents/search_v3.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        # Parse out the action and action input
+        action = parsed["action"]
+        action_input = parsed["action_input"]
+
+        if action == "Tool_Finish":
+            return AgentFinish(return_values={"output": action_input}, log=llm_output)
+        return AgentAction(tool=action, tool_input=action_input, log=llm_output)
+
+
+class ActionRunner:
```
Comment (yukiontheiceberg) [inline autoagents/agents/agents/search_v3.py:111]: Can we inherit this from the v2 ActionRunner?

File: autoagents/eval/test_v3.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import asyncio
```
Comment (rmfan) [inline autoagents/eval/test_v3.py:1]: Do we really need a separate test file for v3?

File: autoagents/agents/agents/search_v3.py (hunk 1 above)
Comment (odp) [reply autoagents/agents/agents/search_v3.py]: No. Removed. Also added few missing things like tools entry to the log and current date to the goal.

File: autoagents/eval/test_v3.py (hunk 3 above)
Comment (odp) [reply autoagents/eval/test_v3.py:1]: Not really, at this point the file is just acting as documentation on how to use prompt v3. The eval MR has done a major refactoring to this file and should just assimilate the prompt v3 changes and remove the file.

File: autoagents/eval/test_v3.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import json
+
+from autoagents.agents.agents.search_v3 import ActionRunner
+from autoagents.agents.models.custom import CustomLLMV3
+
+
+async def work(user_input):
+    outputq = asyncio.Queue()
+    llm = CustomLLMV3(model_name="action_vicuna_7b_withabort_v2_16k", max_tokens=1024)
```
Comment (rmfan) [inline autoagents/eval/test_v3.py:10]: 1024 doesnt seem like a lot for `max_tokens`

Comment (odp) [reply autoagents/eval/test_v3.py:10]: The arguments are from an arbitrary example used for functional testing of code. The end user should change as per the allowed model context size etc.

File: autoagents/agents/agents/search_v3.py (hunk 2 above)
Comment (odp) [reply autoagents/agents/agents/search_v3.py]: We could but at this point v3 ActionRunner has almost all it's subcomponents (which are also interdependent) different from v2 (like tools, prompt, output parser, callback handler etc.) that inheriting from v2 complicates the things, so created a new `ActionRunnerV3`

**Review by rmfan — APPROVED** [review]

**Pull request merged**